builtInFunctions/esdtDeleteMetadata_test.go

```diff
@@ -0,0 +1,245 @@
+
+	args = createMockArgsForNewESDTDelete()
+	e, err := NewESDTDeleteMetadataFunc(args)
+	assert.Nil(t, err)
+	assert.False(t, e.IsInterfaceNil())
+	assert.True(t, e.IsActive())
+
+	args = createMockArgsForNewESDTDelete()
+	args.ActivationEpoch = 1
+	e, _ = NewESDTDeleteMetadataFunc(args)
+	assert.False(t, e.IsActive())
+
+	e.SetNewGasConfig(&vmcommon.GasCost{})
+}
+
+func TestEsdtDeleteMetaData_ProcessBuiltinFunctionErrors(t *testing.T) {
+	t.Parallel()
+
+	args := createMockArgsForNewESDTDelete()
+	e, _ := NewESDTDeleteMetadataFunc(args)
+
+	vmOutput, err := e.ProcessBuiltinFunction(nil, nil, nil)
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrNilVmInput)
+
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, &vmcommon.ContractCallInput{VMInput: vmcommon.VMInput{CallValue: big.NewInt(10)}})
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrBuiltInFunctionCalledWithValue)
+
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, &vmcommon.ContractCallInput{VMInput: vmcommon.VMInput{CallValue: big.NewInt(0)}})
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrAddressIsNotAllowed)
+
+	vmInput := &vmcommon.ContractCallInput{VMInput: vmcommon.VMInput{CallValue: big.NewInt(0)}}
+	vmInput.CallerAddr = e.allowedAddress
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrInvalidRcvAddr)
+
+	vmInput.RecipientAddr = e.allowedAddress
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrInvalidNumOfArgs)
+
+	e.delete = false
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrInvalidNumOfArgs)
+
+	vmInput.Arguments = [][]byte{{1}, {0}, {1}}
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.NotNil(t, err)
+
+	e.delete = true
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrInvalidNumOfArgs)
+
+	vmInput.Arguments = [][]byte{{1}, {0}, {1}, {1}}
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.NotNil(t, err)
+
+	e.delete = false
+
+	acnt := mock.NewUserAccount(vmcommon.SystemAccountAddress)
+	accounts := &mock.AccountsStub{LoadAccountCalled: func(address []byte) (vmcommon.AccountHandler, error) {
+		return acnt, nil
+	}}
+
+	e.accounts = accounts
+	vmInput.Arguments = [][]byte{{1}, {0}, {1}}
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrInvalidNonce)
+
+	vmInput.Arguments[0] = []byte("TOKEN-ABABAB")
+	vmInput.Arguments[1] = []byte{1}
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.Equal(t, err, ErrInvalidTokenID)
+
+	vmInput.Arguments[0] = []byte("TOKEN-ababab")
+	vmOutput, err = e.ProcessBuiltinFunction(nil, nil, vmInput)
+	assert.Nil(t, vmOutput)
+	assert.NotNil(t, err)
```

Comment:
for these `assert.NotNil(t, err)` it is not clear what exactly went wrong. add a comment at least?

Reply:
some unmarshal errors. it does not matter - only for code coverage.